hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -2616,6 +2617,22 @@ public Integer getWritesFileIdEncoding() {
     return props.getInteger(WRITES_FILEID_ENCODING, HoodieMetadataPayload.RECORD_INDEX_FIELD_FILEID_ENCODING_UUID);
   }
 
+  public boolean needResolveWriteConflict(Option<WriteOperationType> operationType) {
+    if (getWriteConcurrencyMode().supportsOptimisticConcurrencyControl()) {
+      // Skip to resolve conflict for non bulk_insert operation if using non-blocking concurrency control
+      // TODO: skip resolve conflict if the option is empty or the inner operation type is UNKNOWN ?
+      return !isNonBlockingConcurrencyControl() || mayBeBulkInsert(operationType);
```

Comment:
Remove the Option from the param and change the logic to:

```java
return  BULK_INSERT.equals(operationType.get()) || !isNonBlockingConcurrencyControl()
```

Reply:
Just to ensure the following two cases:
1. We wanna skip resolve conflict if the option is empty or the inner operation type is UNKNOWN?
2. If the operationType is Option.empty, operationType.get() would throw `NoSuchElementException`. That's what we what? Or using `BULK_INSERT.equals(operationType.orElse(null))`?